### PR TITLE
Add new DTO for updateUserData and add validation for field additionalOrders 

### PR DESCRIPTION
**Pull request by Natalia1910**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: service/src/main/java/greencity/mapping/UserToUserProfileUpdateDto.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class UserToUserProfileUpdateDto extends AbstractConverter<User, UserProfileUpdateDto> {
+    @Override
+    protected UserProfileUpdateDto convert(User user) {
+        List<Address> addresses = user.getAddresses();
+        List<AddressDto> addressDtoList = addresses.stream()
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/UserToUserProfileUpdateDto.java:18]: Is it need to create list of Address? Why not write user.getAddresses .stream()... ?

File: service/src/main/java/greencity/mapping/UserToUserProfileUpdateDto.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            .map(address -> new AddressDto(address.getId(), address.getCity(), address.getDistrict(),
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/UserToUserProfileUpdateDto.java:19]: Will be better if you build this Object.

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 3]
```diff
@@ -859,11 +859,11 @@ public AllPointsUserDto findAllCurrentPointsForUser(String uuid) {
...
-    public UserProfileDto updateProfileData(String uuid, UserProfileDto userProfileDto) {
+    public UserProfileUpdateDto updateProfileData(String uuid, UserProfileUpdateDto userProfileUpdateDto) {
         createUserByUuidIfUserDoesNotExist(uuid);
         User user = userRepository.findByUuid(uuid);
-        setUserData(user, userProfileDto);
-        List<AddressDto> addressDtoList = userProfileDto.getAddressDto();
+        setUserData(user, userProfileUpdateDto);
+        List<AddressDto> addressDtoList = userProfileUpdateDto.getAddressDto();
         List<Address> addressList =
             addressDtoList.stream().map(a -> modelMapper.map(a, Address.class)).collect(Collectors.toList());
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java:868]: same with mapper. Why not write serProfileUpdateDto.getAddressDto().stream()...

**Review by VadymMakitra — CHANGES_REQUESTED** [review]

File: service/src/main/java/greencity/mapping/UserToUserProfileUpdateDto.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import greencity.entity.user.User;
+import greencity.entity.user.ubs.Address;
+import org.modelmapper.AbstractConverter;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class UserToUserProfileUpdateDto extends AbstractConverter<User, UserProfileUpdateDto> {
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/UserToUserProfileUpdateDto.java:14]: Please add test for this mapper.

**Review by Max877866 — APPROVED** [review]

**Review by RossSihovsk — APPROVED** [review]

**Review by thesvider — CHANGES_REQUESTED** [review]
Please resolve comments

**Review by VadymMakitra — APPROVED** [review]

File: service/src/main/java/greencity/mapping/UserToUserProfileUpdateDtoMapper.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class UserToUserProfileUpdateDtoMapper extends AbstractConverter<User, UserProfileUpdateDto> {
+    @Override
+    protected UserProfileUpdateDto convert(User user) {
+        List<AddressDto> addressDtoList = user.getAddresses().stream()
+            .map(address -> AddressDto.builder()
```
Comment (thesvider) [inline service/src/main/java/greencity/mapping/UserToUserProfileUpdateDtoMapper.java:18]: add `private AddressDto createAdressDto(Address)` method

**Review by thesvider — APPROVED** [review]

**Pull request merged**